tfjs-backend-webgpu/src/backend_webgpu.ts

```diff
@@ -840,10 +840,21 @@ export class WebGPUBackend extends KernelBackend {
     });
     this.ensureCommandEncoderReady();
     const passEncoder = this.currentCommandEncoder.beginComputePass();
+    const shouldTimeProgram = this.activeTimers != null;
+    if (shouldTimeProgram) {
+      if (this.supportTimeQuery) {
+        passEncoder.writeTimestamp(this.querySet, 0);
+      }
+    }
     passEncoder.setPipeline(program.pipeline);
     passEncoder.setBindGroup(0, bindGroup);
     passEncoder.dispatch(
         program.dispatch[0], program.dispatch[1], program.dispatch[2]);
+    if (shouldTimeProgram) {
+      if (this.supportTimeQuery) {
+        passEncoder.writeTimestamp(this.querySet, 1);
+      }
+    }
     passEncoder.endPass();
```

Comment:
Can you move `activeTimers.push` here instead of a separating file?
```
  if (shouldTimeProgram) {
    this.activeTimers.push({
      name: program.constructor.name,
      query: getQueryTime(this.querySet)
    });
  }

Reply:
Updated. And also use threshold `WEBGPU_DEFERRED_SUBMIT_BATCH_SIZE` in FromPixels to control queue submit timing.